Table.read() error when reading ECSV using astropy dev

_Issue [JP-2068](https://jira.stsci.edu/browse/JP-2068) was created on JIRA by [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies):_

We are seeing a failure in our weekly cron job of unit tests using dev dependencies.  This one is because of some changes to the ECSV format in astropy dev:

<https://github.com/spacetelescope/jwst/runs/2489783314?check_suite_focus=true#step:9:51414>

The failing tests in the `jwst` repo are:


```
test_calc_deltas
test_from_models
```


failing with the same error in `Table.read()` when reading an ECSV file:


```
E               ValueError: datatype 'object' of column 'exposure' is not in allowed values ('bool', 'int8', 'int16', 'int32', 'int64', 'uint8', 'uint16', 'uint32', 'uint64', 'float16', 'float32', 'float64', 'float128', 'string')
```

Also, the test should be using a context manager when opening the ECSV file using `Table.read()`, hence all the open-files ERRORs, though clearly these 2 tests are contaminating others, somehow.

--- a/jwst/lib/tests/data/calc_deltas_truth.ecsv
+++ b/jwst/lib/tests/data/calc_deltas_truth.ecsv
@@ -1,7 +1,7 @@
 # %ECSV 0.9
 # ---
 # datatype:
-# - {name: exposure, datatype: object}
+# - {name: exposure, datatype: string}
 # - {name: target.ra, unit: deg, datatype: float64}
 # - {name: target.dec, unit: deg, datatype: float64}
 # - {name: v1.ra, unit: deg, datatype: float64}

--- a/jwst/lib/tests/data/v1_calc_truth.ecsv
+++ b/jwst/lib/tests/data/v1_calc_truth.ecsv
@@ -1,7 +1,7 @@
 # %ECSV 0.9
 # ---
 # datatype:
-# - {name: source, datatype: object}
+# - {name: source, datatype: string}
 # - {name: obstime, datatype: string}
 # - {name: ra, datatype: float64}
 # - {name: dec, datatype: float64}

--- a/jwst/lib/tests/test_pointing_summary.py
+++ b/jwst/lib/tests/test_pointing_summary.py
@@ -7,7 +7,7 @@ import pytest
 from astropy.table import Table
 from astropy.utils.diff import report_diff_values
 
-import jwst.datamodels as dm
+from jwst.datamodels import ImageModel
 from jwst.lib import engdb_tools
 import jwst.lib.pointing_summary as ps
 
@@ -24,14 +24,13 @@ GOOD_ENDTIME = '2016-01-19'
 def engdb():
     """Setup the service to operate through the mock service"""
     with EngDB_Mocker():
-        engdb = engdb_tools.ENGDB_Service()
-        yield engdb
+        yield engdb_tools.ENGDB_Service()
 
 
 @pytest.fixture(scope='module')
 def data_path(jail):
     """Create data file with needed header parameters"""
-    model = dm.ImageModel()
+    model = ImageModel()
 
     model.meta.target.ra = 90.75541666666666
     model.meta.target.dec = -66.56055555555554
@@ -52,16 +51,16 @@ def test_calc_pointing_deltas(engdb, data_path):
              '\n    (90.70377653, -66.59540224)>, delta_v1=<Angle 0.13712727'
              ' deg>, delta_refpoint=<Angle 0.04044315 deg>)'
              )
-    model = dm.ImageModel(str(data_path))
-    deltas = ps.calc_pointing_deltas(model)
+    with ImageModel(str(data_path)) as model:
+        deltas = ps.calc_pointing_deltas(model)
 
     assert truth == str(deltas)
 
 
 def test_calc_deltas(engdb, data_path):
     """Test `calc_deltas` basic running"""
-    model = dm.ImageModel(data_path)
-    deltas = ps.calc_deltas([model])
-    truth = Table.read(DATA_PATH / 'calc_deltas_truth.ecsv')
+    with ImageModel(data_path) as model:
+        deltas = ps.calc_deltas([model])
 
+    truth = Table.read(DATA_PATH / 'calc_deltas_truth.ecsv')
     assert report_diff_values(truth, deltas, fileobj=sys.stderr)

--- a/jwst/lib/tests/test_v1_calculate.py
+++ b/jwst/lib/tests/test_v1_calculate.py
@@ -8,7 +8,7 @@ from astropy.table import Table
 from astropy.time import Time
 from astropy.utils.diff import report_diff_values
 
-import jwst.datamodels as dm
+from jwst.datamodels import ImageModel
 from jwst.lib import engdb_tools
 import jwst.lib.v1_calculate as v1c
 
@@ -25,13 +25,12 @@ GOOD_ENDTIME = '2016-01-19'
 def engdb():
     """Setup the service to operate through the mock service"""
     with EngDB_Mocker():
-        engdb = engdb_tools.ENGDB_Service()
-        yield engdb
+        yield engdb_tools.ENGDB_Service()
 
 
 def test_from_models(engdb):
     """Test v1_calculate_from_models for basic running"""
-    model = dm.ImageModel()
+    model = ImageModel()
     model.meta.exposure.start_time = Time(GOOD_STARTTIME).mjd
     model.meta.exposure.end_time = Time(GOOD_ENDTIME).mjd
 
@@ -39,7 +38,6 @@ def test_from_models(engdb):
     v1_formatted = v1c.simplify_table(v1_table)
 
     truth = Table.read(DATA_PATH / 'v1_calc_truth.ecsv')
-
     assert report_diff_values(truth, v1_formatted, fileobj=sys.stderr)
 
 
@@ -51,5 +49,4 @@ def test_over_tiome(engdb):
     v1_formatted = v1c.simplify_table(v1_table)
 
     truth = Table.read(DATA_PATH / 'v1_time_truth.ecsv')
-
     assert report_diff_values(truth, v1_formatted, fileobj=sys.stderr)

